### PR TITLE
Container doc rewrite

### PR DESCRIPTION
**Pull request by micwade-aws**
*Description:*
Large rewrite of container documentation to clarify usage of DLC and removal of sidecar. The bulk of the information is now available as suggested "developer workflows".

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: neuron-deploy/k8.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.. _self-managed-kubernetes-service:
+
+Self Managed Kubernetes Service
+===============================
+Introduction
+------------
+Use of Neuron in containers on Kubernetes cluster can be simple to achieve by following :ref:`tutorial-k8s-env-setup-for-neuron`
```
Comment (AWSGH) [inline neuron-deploy/k8.rst:7]: "...on a Kubernetes cluster.."

**Review by aws-maens — APPROVED** [review]

File: neuron-deploy/sm_notebook.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.. _sagemaker-notebook:
+
+Sagemaker Notebook
```
Comment (AWSGH) [inline neuron-deploy/sm_notebook.rst:3]: their official name is "Amazon SageMaker Studio Notebooks"

File: neuron-deploy/sm_notebook.rst [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+==================
+Introduction
+------------
+Use of Neuron with Sagemaker Notebook can be simple to achieve by following this
+    - PytorchNotebook_
```
Comment (AWSGH) [inline neuron-deploy/sm_notebook.rst:8]: PyTorch

File: neuron-deploy/sm_notebook.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    - TensorflowNotebook_
```
Comment (AWSGH) [inline neuron-deploy/sm_notebook.rst:9]: TensorFlow

File: neuron-deploy/sm_notebook.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+.. _PytorchNotebook: https://github.com/aws/amazon-sagemaker-examples/tree/master/sagemaker_neo_compilation_jobs/deploy_pytorch_model_on_Inf1_instance
+.. _TensorflowNotebook: https://github.com/aws/amazon-sagemaker-examples/tree/master/sagemaker_neo_compilation_jobs/deploy_tensorflow_model_on_Inf1_instance
+
+Debug Hints
+-----------
+Sagemaker endpoints have cloudwatch metrics. For more details please check `here <https://docs.aws.amazon.com/sagemaker/latest/dg/monitoring-cloudwatch.html>`_
```
Comment (AWSGH) [inline neuron-deploy/sm_notebook.rst:16]: replace with "You can Monitor Amazon SageMaker with Amazon CloudWatch. For more details...."

File: neuron-deploy/tutorials/k8s_rn50_demo.rst [hunk 6]
```diff
@@ -0,0 +1,193 @@
+.. _example-deploy-rn50-as-k8s-service:
+
+Example: Deploy Resnet50 as a k8s service
```
Comment (AWSGH) [inline neuron-deploy/tutorials/k8s_rn50_demo.rst:3]: "Deploy a TensorFlow Resnet50 model as a k8s service"

File: neuron-deploy/tutorials/k8s_rn50_demo.rst [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+-------------------------------------
+
+This tutorial uses Resnet50 model as a teaching example on how to deploy an
+inference application using Kubernetes on the Inf1 instances.
+
+Prerequisite:
```
Comment (AWSGH) [inline neuron-deploy/tutorials/k8s_rn50_demo.rst:9]: Prerequisites

File: neuron-deploy/tutorials/k8s_rn50_demo.rst [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+Deploy a TensorFlow Serving application image
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A trained model must be compiled to an Inferentia target before it can be deployed on Inferentia instances\.
+To continue, you will need a `Neuron optimized TensorFlow <https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-guide/neuron-frameworks/tensorflow-neuron/index.html>`_ model saved in Amazon S3\.
+If you don’t already have a SavedModel, please follow the tutorial for `creating a Neuron compatible ResNet50 model <https://docs.aws.amazon.com/dlami/latest/devguide/tutorial-inferentia-tf-neuron.html>`_
+and upload the resulting SavedModel to S3\.
+
+ResNet-50 is a popular machine learning model used for image
+recognition tasks\. For more information about compiling Neuron models, see
```
Comment (AWSGH) [inline neuron-deploy/tutorials/k8s_rn50_demo.rst:29]: "image classification"

File: neuron-deploy/tutorials/k8s_rn50_demo.rst [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+`The AWS Inferentia Chip With DLAMI <https://docs.aws.amazon.com/dlami/latest/devguide/tutorial-inferentia.html>`_
```
Comment (AWSGH) [inline neuron-deploy/tutorials/k8s_rn50_demo.rst:30]: lets not point to the DLAMI docs, stay within RTD for Neuron documentation.

File: neuron-intro/devflows/dlc-then-eks-devflow.rst [hunk 10]
```diff
@@ -12,19 +12,82 @@ Deploy DLC on EKS
...
 
+|image|
+ 
+.. |image| image:: /images/dlc-on-eks-dev-flow.png
+   :width: 750
+   :alt: Neuron developer flow for DLC on ECS
+   :align: middle
 
-Placeholder: Diagram
+You can use the Neuron version of the `AWS Deep Learning Containers <https://docs.aws.amazon.com/deep-learning-containers/latest/devguide/deep-learning-containers-ecs-tutorials-inference.html>`_ to run inference on Amazon Elastic Kubernetes Service (EKS). In this developer flow, you set up an EKS cluster with inf1 instances, create a Kubernetes manifest for your inference service and deploy it to your cluster. This developer flow assumes:
```
Comment (AWSGH) [inline neuron-intro/devflows/dlc-then-eks-devflow.rst:22]: Inf1

**Review by AWSGH — APPROVED** [review]
please address minor lang and convention requests

**Comment by micwade-aws** [comment]
> please address minor lang and convention requests

done.  moving forward with the merge.

**Pull request merged**